### PR TITLE
8300794: Use @snippet in java.util:i18n

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this javadoc only change which uses `@snippet` and `@linkplain` in i18n related util packages.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300794](https://bugs.openjdk.org/browse/JDK-8300794): Use @snippet in java.util:i18n

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [e56955e1](https://git.openjdk.org/jdk/pull/13920/files/e56955e1d29ff95eb68c30cdbd97251c379920cf)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**) ⚠️ Review applies to [e56955e1](https://git.openjdk.org/jdk/pull/13920/files/e56955e1d29ff95eb68c30cdbd97251c379920cf)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13920/head:pull/13920` \
`$ git checkout pull/13920`

Update a local copy of the PR: \
`$ git checkout pull/13920` \
`$ git pull https://git.openjdk.org/jdk.git pull/13920/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13920`

View PR using the GUI difftool: \
`$ git pr show -t 13920`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13920.diff">https://git.openjdk.org/jdk/pull/13920.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13920#issuecomment-1542953271)

### CODE REVIEW

File: src/java.base/share/classes/java/util/Currency.java [hunk 1]
```diff
@@ -514,7 +514,7 @@ public String getCurrencyCode() {
      * <p>
      * If the default {@link Locale.Category#DISPLAY DISPLAY} locale
      * contains "rg" (region override)
-     * <a href="./Locale.html#def_locale_extension">Unicode extension</a>,
+     * {@linkplain Locale##def_locale_extension Unicode extensions},
```
Comment (naotoj) [inline src/java.base/share/classes/java/util/Currency.java:517]: Good catch on plurals

**Review by naotoj — COMMENTED** [review]
Looks good overall. Some minor comments follow.

File: src/java.base/share/classes/java/util/ResourceBundle.java [hunk 2]
```diff
@@ -512,9 +508,9 @@ public ResourceBundle() {
     /**
      * Gets a string for the given key from this resource bundle or one of its parents.
      * Calling this method is equivalent to calling
-     * <blockquote>
-     * <code>(String) {@link #getObject(java.lang.String) getObject}(key)</code>.
-     * </blockquote>
+     * {@snippet lang=java :
+     *     (String[]) getObject(key); // @link substring="getObject" target="#getObject(java.lang.String)"
```
[the remark this answers is not in the document]
Comment (justin-curtis-lu) [reply src/java.base/share/classes/java/util/ResourceBundle.java]: Thanks for the review, moved the links on top, and fixed the indentation in the example snippet.

**Review by naotoj — APPROVED** [review]

**Pull request closed without merging**